### ISSUE TITLE
`foo'` links to foo

### ISSUE THREAD

**Issue by zliu41** [haskell/haddock#1408]
While `'foo''` and ``` `foo'' ``` both correctly link to `foo'`, ``` `foo'` ``` links to `foo`, which seems like a bug.

**Comment by Kleidukos** [comment]
Alright, I can reproduce it with the following code:

```haskell
module Module where

-- | This is the `foo'` function.
foo' :: ()
foo' = ()
```

```bash
❯ haddock Module.hs 
Haddock coverage:
Warning: Package name is not available.
Warning: 'foo' is out of scope.
    If you qualify the identifier, haddock can try to link it anyway.
  50% (  1 /  2) in 'Module'
  Missing documentation for:
    Module header
```

**Comment by Kleidukos** [comment]
closed by #1483
